yuca/wrappers/glider_wrapper.py

```diff
@@ -94,37 +97,44 @@ def meta_step(self, action):
                 
 
 #                global y_displacement_0
-                y_displacement_1 = (self.y_grid * action).sum() / (eps + action.sum())
+                y_displacement_1 = (self.y_grid * action[0:1]).sum() / (eps + action.sum())
+
 
                 # penalize growth/decay, i.e. incentivize morphological homeostasis
                 #
-                growth = 1.0 - (mean_grid / (mean_grid_0 + eps))
-                mean_grid_0 = mean_grid
-                mean_grid = action.mean()
+                growth = 0.0 * torch.abs(1.0 - (mean_grid / (mean_grid_0 + eps)))
+                mean_grid_0 = 1.0 * mean_grid
 
                 if y_displacement_0 != 0.0: #None:
-                    y_displacement = torch.abs(y_displacement_1 - y_displacement_0)
-                    reward += torch.abs(y_displacement.cpu()) - torch.abs(growth.cpu()) * 150
+
+                    y_displacement = y_displacement_1 - y_displacement_0
+
+                    if y_displacement_0 > (kernel_span - 2) and y_displacement_1 < 2:
+                        # account for wraparound errors
+                        y_displacement += kernel_span
+
+                    reward += y_displacement.cpu() - growth.cpu() #* 150
                 else: 
                     y_displacement = torch.tensor(0.0)
-                    reward += torch.abs(y_displacement.cpu()) - torch.abs(growth.cpu()) * 150
+                    reward += y_displacement.cpu() - growth.cpu() #* 150
 
                 y_displacement_0 = y_displacement_1.clone()
 
+
             dgrid = action - old_grid
             old_grid = 1.0  * action
 
             if mean_grid == 0.0 or dgrid.max() <= 0.00001:
                 break
 
-        if step == 0:
-            # empty patterns are extremely penalized 
-            reward -= 10000
-        if mean_grid == 0.0 or dgrid.max() <= 0.00001:
-            # patterns that die out are substantially penalized 
-            reward -= 100
+#        if step == 0:
+#            # empty patterns are extremely penalized 
+#            reward -= 10000
+#        if mean_grid == 0.0 or dgrid.max() <= 0.00001:
+#            # patterns that die out are substantially penalized 
+#            reward -= 100
         
-        reward -= (self.ca_steps - step)
```

Comment:
It _seems_ like we should be able to leave this part out, because agents will get more rewards for longer run times naturally. In practice there seems to be a fine line between inadvertently selecting for long-lived static patterns and transient mobile ones.